R/dataset-stream.R

```diff
@@ -0,0 +1,59 @@
+#' Tools for working with streaming data
+#'
+#' Crunch allows you to stream data to a dataset. Streaming data is useful for
+#' datasets which have frequent updates (see 
+#' [the Crunch documentation](http://docs.crunch.io/#streaming-rows) for more)
+#' information. `pendingMessages()` retrieves the number of pending messages and `appendStreamedRows()` appends all pending streamed rows to the dataset.
+#'
+#' @param x a CrunchDataset
+#' @param ... Additional batch attributes to set when appending pending stream 
+#' data. Options include
+#' `name` and `description`.
+#' @return number of pending messages for `pendingMessages()` and the dataset 
+#' with appended messages `appendStreamedRows()`
+#' @name streaming
+#' @aliases streaming pendingMessages appendStreamedRows
+#' @examples
+#' \dontrun{
+#' # need examples!
+#' }
+NULL
+
+
+#' @rdname streaming
+#' @export
+setMethod("pendingMessages", "CrunchDataset", function (x) {
```

Comment:
I went back and forth myself, I'll bump them down to just functions for now. If we want to implement a stream from R where you can do something like `stream(ds) <- [some data frame]` then we can align all of these together, but for now it's probably more than is necessary.